src/asio_service.cxx

```diff
@@ -1226,6 +1256,9 @@ class asio_rpc_client
             if (impl_->get_options().replicate_log_timestamp_) {
                 ss.put_u64( le->get_timestamp() );
             }
+            if (impl_->get_options().crc_on_payload_) {
+                ss.put_u32(le->get_crc32());
+            }
```

Comment:
Similarly, we need to ship the existence flag as well.
```c++
if (impl_->get_options().crc_on_payload_) {
    ss.put_u8(le->has_crc32() ? 1 : 0);
    ss.put_u32(le->get_crc32());
}
```
and
```c++
LOG_ENTRY_SIZE += 5;
```
I don't have better idea rather than shipping 5 bytes.